DEVELOPMENT.md

```diff
@@ -1,77 +1,138 @@
+[deploy your own version of apps plugin](#starting-apps-plugin)!
 
-```sh
-tanzu init
-```
+Before submitting a PR, see also [CONTRIBUTING.md](./CONTRIBUTING.md).
 
-```sh
-tanzu plugin repo add -b tanzu-cli-admin-plugins -n admin -p artifacts-admin
-```
+### Install requirements
 
-```sh
-tanzu plugin install builder
+You must install these tools:
+
+- [`go`]: for compiling the plugin as well as other dependencies - 1.16+
+- [`tanzu`]: for adding the plugin
+- [`git`](https://help.github.com/articles/set-up-git/): for source control
+- [`tanzu builder plugin`] for compiling and publishing plugins
+
+
+[`go`]: https://golang.org/doc/install
+[`tanzu`]: https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/cli/getting-started.md#install-the-latest-release-of-tanzu-cli
+
+### Check out your fork
+
+To check out this repository:
+
+1. Create your own
+   [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine:
+
+```shell
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/apps-cli-plugin.git
+cd apps-cli-plugin
+git remote add upstream https://github.com/vmware-tanzu/apps-cli-plugin.git
+git remote set-url --push upstream no_push
 ```
 
+_Adding the `upstream` remote sets you up nicely for regularly
+[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+
+Once you reach this point you are ready to do a full build and install as
+described below.
+
+## Building
+
+Once you've [set up your development environment](#prerequisites), let's build
+`apps plugin`.
+
+**Dependencies:**
+
+[go mod](https://github.com/golang/go/wiki/Modules#quick-start) is used and
+required for dependencies.
+#### Install builder plugin
+
+Install the Tanzu builder and test plugins. Set variable `OS` to `linux`, `darwin` or `windows` depending on your OS:
+
 ```sh
-tanzu plugin install test
+TANZU_VERSION=$(cat TANZU_VERSION)
+TANZU_HOME=$HOME/tanzu
+curl -Lo admin-plugins.tar.gz https://github.com/vmware-tanzu/tanzu-framework/releases/download/${TANZU_VERSION}/tanzu-framework-plugins-admin-${OS}-amd64.tar.gz
+tar -xzf admin-plugins.tar.gz -C ${TANZU_HOME}
+tanzu plugin install builder --local ${TANZU_HOME}/admin-plugins
+tanzu plugin install test --local ${TANZU_HOME}/admin-plugins
 ```
 
+**Building:**
+
 To build and install the apps plugin, run: (repeat this step any time you pull new source code to get the latest)
 
 ```sh
 make install
 ```
 
-## Command documentation
+## Testing
 
-Command argument and flag documentation is available in the [docs](./docs/tanzu_apps.md) directory. CI will fail if this content is not kept up to date.
+### Unit testing
 
-To regenerate the docs, run:
+All unit tests can be run on any machine with go 1.17+ installed.
 
 ```sh
-make docs
+make test
 ```
 
-## Unit testing
+### Acceptance testing
 
-All unit tests can be run on any machine with go 1.16+ installed.
+In order to use the CLI, the runtime dependencies need to be installed into the target Kubernetes cluster.
```

Comment:
`...installed into the target Kubernetes...`

maybe better if it read as:

`...installed on the target Kubernetes...`